Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Base class Interactable GameObjects can inherit from.
+    /// </summary>
+    public class Interactable : MonoBehaviour
+    {
+        protected bool TriggerKeywordOnGazeOnly;
+
+        protected bool IsFocused;
+
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized += KeywordRecognized;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized -= KeywordRecognized;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any gameObject.
+        /// </summary>
+        /// <param name="tappedGameObject">GameObject user has tapped.</param>
+        protected virtual void OnTap(GameObject tappedGameObject)
+        {
+            if (tappedGameObject == gameObject)
```

Comment:
The time could be reset when it prints.  It's useful to accumulate multiple samples to reduce the impact of outliers.  It's similar to the test you provided, but my 'reps' are done in successive frames rather than in a loop, so I can't use one sample of time.

True, the events won't be called every frame.  They won't be called ten million times, either.  :) It's just a useful way to get a lot of samples in a short amount of time.

The timer could be moved to after the random value.  All of the classes have the same behavior though, so that is unlikely to change how the results compare with one another.